### ISSUE TITLE
Sometimes crash on Mac M1

### ISSUE THREAD

**Issue by scheglov** [dart-lang/sdk#49630]
```console
scheglov@scheglov-macbookpro4:~/Source/Dart/sdk.git/sdk (aa-publish)$ d-rts
Total 447 tests, of which 104 are expected to fail.
00:53 +4962: analysis_server | lsp | CompletionTest | test_completeFunctionCalls_resolve_producesCorrectEditWithoutInsertText
===== CRASH =====
si_signo=Segmentation fault: 11(11), si_code=2, si_addr=0x0
version=2.18.0-edge.a37502c1b88a7fa3277531b37a5b075169a8f9c1 (be) (Tue Jun 28 16:02:53 2022 +0000) on "macos_arm64"
pid=57064, thread=20999, isolate_group=main(0x12700c400), isolate=main(0x12701bc00)
isolate_instructions=10425f7e0, vm_instructions=10425f7e0
Stack dump aborted because InitialRegisterCheck failed.
/Users/scheglov/bin/d-rts: line 11: 57064 Abort trap: 6           dart /Users/scheglov/Source/Dart/sdk.git/sdk/pkg/analysis_server/test/test_all.dart

real	1m3.397s
user	1m11.088s
sys	0m5.688s
```

```console
scheglov@scheglov-macbookpro4:~/Source/Dart/sdk.git/sdk (aa-publish)$ dart --version
Dart SDK version: 2.18.0-edge.a37502c1b88a7fa3277531b37a5b075169a8f9c1 (be) (Tue Jun 28 16:02:53 2022 +0000) on "macos_arm64"
scheglov@scheglov-macbookpro4:~/Source/Dart/sdk.git/sdk (aa-publish)$ uname -a
Darwin scheglov-macbookpro4.roam.internal 21.6.0 Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:22 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T6000 arm64
```

I think this happens most often when I run this set of tests in parallel with another one or two tests in different console tabs.

**Comment by a-siva** [comment]
//cc @rmacnak-google

**Comment by rmacnak-google** [comment]
@scheglov Can you expand what your `d-rts` script runs?

**Comment by scheglov** [comment]
```console
time dart /Users/scheglov/Source/Dart/sdk.git/sdk/pkg/analysis_server/test/test_all.dart
```

**Comment by a-siva** [comment]
possibly on out of memory situation in some C code that does not have a check.

**Comment by rmacnak-google** [comment]
I haven't been able to reproduce this. Are you able to reproduce this with a coredump or under lldb?

**Comment by a-siva** [comment]
@scheglov any luck on reprocing this with a coredump ?

**Comment by scheglov** [comment]
I have not seen it for a while.

**Comment by a-siva** [comment]
Closing issue, if you run into this again please open a new issue.
